tests/python/pytorch/graphbolt/impl/test_sampled_subgraph_impl.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+import torch
+from dgl.graphbolt.impl.sampled_subgraph_impl import (
+    exclude_edges,
+    SampledSubgraphImpl,
+)
+
+
+def _assert_container_equal(lhs, rhs):
+    if isinstance(lhs, torch.Tensor):
+        assert isinstance(rhs, torch.Tensor)
+        assert torch.equal(lhs, rhs)
+    elif isinstance(lhs, tuple):
+        assert isinstance(rhs, tuple)
+        assert len(lhs) == len(rhs)
+        for l, r in zip(lhs, rhs):
+            _assert_container_equal(l, r)
+    elif isinstance(lhs, dict):
+        assert isinstance(rhs, dict)
+        assert len(lhs) == len(rhs)
+        for key, value in lhs.items():
+            assert key in rhs
+            _assert_container_equal(value, rhs[key])
+
+
+@pytest.mark.parametrize("is_reverse", [True, False])
+def test_exclude_edges_homo(is_reverse):
+    node_pairs = (torch.tensor([0, 2, 3]), torch.tensor([1, 4, 2]))
+    if is_reverse:
+        reverse_row_node_ids = torch.tensor([10, 15, 11, 24, 9])
+        reverse_column_node_ids = torch.tensor([10, 15, 11, 24, 9])
+    else:
+        reverse_row_node_ids = reverse_column_node_ids = None
+    reverse_edge_ids = torch.Tensor([5, 9, 10])
+    subgraph = SampledSubgraphImpl(
+        node_pairs,
+        reverse_column_node_ids,
+        reverse_row_node_ids,
+        reverse_edge_ids,
+    )
+    if is_reverse:
+        excluded_edges = (torch.tensor([11]), torch.tensor([9]))
+    else:
+        excluded_edges = (torch.tensor([2]), torch.tensor([4]))
```

Comment:
by any chance user will use compacted edge id?

Reply:
Yeah. This should be removed.

Reply:
I found the `if` branch cannot be removed in the tests. For the case of `is_reverse == False`, it means the subgraph is not compacted and the edge `(2, 4)` means an edge between original ids.